cloudmarker/alerts/slackalert.py

```diff
@@ -0,0 +1,108 @@
+"""Alerter to send Slack messages for identified anomalies."""
+
+import json
+import logging
+
+import slackclient
+
+_log = logging.getLogger(__name__)
+
+
+class SlackAlert:
+    """Alert plugin to send Slack notifications."""
+
+    def __init__(self, bot_user_token, to, text,
+                 temp_file='/tmp/cloudmarker/slackanomalyreport.json'):
+        """Initialize the class:`SlackAlerter`.
+
+        Arguments:
+          bot_user_token (string): Token for Slack bot user.
+          to (list): List of recipients(string) to send Slack alert to.
+          text (string): Message body.
+          temp_file (string): Name of file to be used to save intrim JSON
+            record which will be used to attach as report to Slack message.
+
+        """
+        self._bot_user_token = bot_user_token
+        self._to = to
+        self._text = text
+        self._temp_file = temp_file
+
+        self._slack_report = []
+
+    def _post_message(self):
+        """Send slack notifications with attachment."""
+        sc = slackclient.SlackClient(self._bot_user_token)
+
+        recipients = self._to
+
+        for recipient in recipients:
+            try:
+                user_id = sc.api_call('users.lookupByEmail',
+                                      email=recipient)['user']['id']
+
+                # Open direct message channel to the user.
+                channel_id = sc.api_call('im.open',
+                                         user=user_id)['channel']['id']
+
+                sc.api_call(
+                    'chat.postMessage',
+                    channel=channel_id,
+                    attachments=[
+                        {
+                            'fallback': 'Cloudmarker Notification!',
+                            'color': '#36a64f',
+                            'title': 'Cloudmarker Anomaly Notification'
```

Comment:
A couple of comments here:

- Let the title just be `Cloudmarker Alert`.
- Search for "notification" or "Notification" in the entire file and change it to "alert" or "Alert". The reason is that we are settling on the term "alert" and we will use it consistently everywhere.
- Please add a comment above that explains how the `fallback` and `title` options are used or what they mean.